code/modules/transport/pods/warp_travel.dm

```diff
@@ -144,4 +173,182 @@ var/global/list/warp_beacons = list() //wow you should've made one for warp beac
+		playsound(get_turf(src), "sound/machines/heater_off.ogg", 20, 1)
+		if(src.beaconid)
+			packitup.beaconid = src.beaconid
+			packitup.name = "warp buoy unit [beaconid]"
+		SPAWN_DBG(0)
+			qdel(src)
+
+//deployable warp beacon
+
+/obj/beacon_deployer
+	name = "warp buoy unit"
+	desc = "A compact anchor for teleportation technology, held together by cut-rate construction supplies. What could possibly go wrong?"
+	icon = 'icons/obj/ship.dmi'
+	icon_state = "beaconunit"
+	density = 1
+	var/deploying = null
+	var/beaconid = null
+
+	New()
+		src.beaconid = rand(1000,9999)
+		src.name = "warp buoy unit [beaconid]"
+		..()
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (istype(W, /obj/item/wrench) && !src.deploying)
+			for (var/turf/a in range(2,src))
+				if (!a.allows_vehicles)
+					boutput(usr,"<span style=\"color:red\">The area surrounding the beacon isn't sufficiently navigable for vehicles.</span>")
+					return
+			if (isrestrictedz(src.z))
+				boutput(usr, "<span style=\"color:red\">The beacon can't connect to the warp network.</span>")
+				return
+//additional check, for if this thing gets used terribly somehow
+//			for (var/area/station/a in orange(1,src))
+//				boutput(user,"<span style=\"color:red\">Interference from nearby electrical fields is preventing this beacon's deployment.</span>")
+//				return
+			src.visible_message("<b>[user.name]</b> deploys [src].")
+			playsound(get_turf(src), "sound/items/Ratchet.ogg", 40, 1)
+			src.deploying = 1
+			src.deploybeacon()
+
+		else if (istype(W, /obj/item/device/multitool/) && !src.deploying)
+			var/str = input(usr,"Set designation","Re-Designate Buoy","") as null|text
+			if (!str || !length(str))
+				boutput(usr, "<span style=\"color:red\">No valid input detected.</span>")
+				return
+			if (length(str) > 30)
+				boutput(usr, "<span style=\"color:red\">Text too long.</span>")
+				return
+			src.beaconid = "[str]"
+			src.name = "warp buoy unit [beaconid]"
+			boutput(usr, "<span style=\"color:blue\">Designation updated to 'Buoy [str]'.</span>")
+		else
+			..()
+
+/obj/beacon_deployer/proc/deploybeacon()
+	src.icon_state = "beacondeploy"
+	src.anchored = 1
+	SPAWN_DBG(16)
+		var/obj/warp_beacon/depbeac = new /obj/warp_beacon/deployed(src.loc)
+		playsound(get_turf(src), "sound/machines/heater_off.ogg", 20, 1)
+		depbeac.name = "Buoy [src.beaconid]"
+		depbeac.beaconid = src.beaconid
+		SPAWN_DBG(0)
+			qdel(src)
+
+/obj/beaconkit
+	name = "warp buoy frame"
+	desc = "A partially completed frame for a deployable warp buoy. It's missing rods for its stand."
+	icon = 'icons/obj/ship.dmi'
+	icon_state = "beacframe_1"
+	density = 1
+	var/state = 1
 
+	attackby(var/obj/item/I as obj, var/mob/user as mob)
+		switch(state)
+			if(1)
+				if (I.amount < 4)
+					boutput(user, "<span style=\"color:red\">You don't have enough rods to complete the stand (4 required).</span>")
+				else if (istype(I, /obj/item/rods))
```

Comment:
checking for rods should probably come before the amount check

Reply:
Resolved with 693bfd0